### PR TITLE
Guidelines for performance considerations with Fortran

### PR DESCRIPTION
**Pull request by rafmudaf**
**Feature or improvement description**
This pull request adds information regarding performance considerations in Fortran. The important content has been added, but the whole page requires a restructure to a more general format before merging.

**Impacted areas of the software**
Documentation

**Additional supporting information**
Live preview at https://raf-openfast.readthedocs.io/en/docs-performance/source/dev/performance.html#high-performance-programming-in-fortran

### CODE REVIEW

File: docs/source/dev/performance.rst [hunk 1]
```diff
@@ -1,16 +1,18 @@
 Performance-Profiling and Optimization
 ======================================
-The OpenFAST team has been engaged in performance-profiling and optimization
-work in an effort to improve the time-to-solution performance for the most
-computationally expensive use cases. This work is supported by Intel® through
-its designation of NREL as an
+A major focus of he OpenFAST team is performance-profiling and optimization
```
Comment (jjonkman) [inline docs/source/dev/performance.rst:3]: Change "he" to "the".

File: docs/source/dev/performance.rst [hunk 2]
```diff
@@ -221,3 +223,170 @@ areas:
...
+alone are not a good indication for the compiler's ability to optimize
+a particular portion of code. Therefore, developers should generate
+optimization reports to get line-by-line reporting on optimizations
+such as vectorization, parallelization, memory and cache usage,
+threading, and others.
+
+https://software.intel.com/content/www/us/en/develop/articles/vectorization-and-optimization-reports.html
+
+- Link to flags for optimization report settings
+- Detail the process for generating the optrpt
```
Comment (jjonkman) [inline docs/source/dev/performance.rst:252]: Are these bullets placeholders for future details?

File: docs/source/dev/performance.rst (hunk 1 above)
Comment (jjonkman) [inline docs/source/dev/performance.rst:3]: Are you going to add in this section any of the profiling results you found regarding linearization within WEIS-OpenFAST?  E.g., where are the computational bottlenecks in the linearization process--when computing the Jacobians of specific modules or when combining the Jacobians of each model into full-system-level Jacobians?

**Review by jjonkman — COMMENTED** [review]
Thanks, @rafmudaf; this looks like valuable information.  Please find must a few minor edits and comments in my review.

File: docs/source/dev/performance.rst (hunk 2 above)
Comment (rafmudaf) [reply docs/source/dev/performance.rst:252]: Yes, I've marked this pull request as a draft. There's still a bit of work to do.

**Review by jjonkman — APPROVED** [review]
I had reviewed this documentation in May and the recent items added look good to me.  I approve the merge.

Thanks!

**Pull request merged**